src/config.cpp

```diff
@@ -76,18 +77,26 @@ static inline void print_availability(std::ostream& os, const std::string& descr
         {
             continue;
         }
-        else if (ev.availability == perf::Availability::PROCESS_MODE)
+
+        std::string availability_str = "";
+        std::string cpu_str = "";
+        if (ev.availability == perf::Availability::PROCESS_MODE)
         {
-            event_names.push_back(ev.name + " *");
+            availability_str = " *";
         }
         else if (ev.availability == perf::Availability::SYSTEM_MODE)
         {
-            event_names.push_back(ev.name + " #");
+            availability_str = " #";
         }
-        else
+        if (ev.get_supported_cpus() != Topology::instance().cpus())
         {
-            event_names.push_back(ev.name);
+            std::set<Cpu> cpus = ev.get_supported_cpus();
```

Comment:
```suggestion
            const auto& cpus = ev.get_supported_cpus();
```
Why copy? Also, `auto` is a thing.